src/main/java/dk/cngroup/lentils/service/ScoreService.java

```diff
@@ -0,0 +1,44 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.Team;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+public class ScoreService {
+
+    private final TeamService teamService;
+    private final StatusService statusService;
+    private final HintTakenService hintTakenService;
+    private final CypherService cypherService;
+
+    @Autowired
+    public ScoreService(final TeamService teamService, final StatusService statusService,
+                        final HintTakenService hintTakenService, final CypherService cypherService) {
+        this.teamService = teamService;
+        this.statusService = statusService;
+        this.hintTakenService = hintTakenService;
+        this.cypherService = cypherService;
+    }
+
+    public int getScoreByTeamAndCypher(final Team team, final Cypher cypher) {
+
+        int hintScore = hintTakenService.getHintScore(team, cypher);
+        int statusScore = statusService.getStatusScore(team, cypher);
+        return (statusScore - hintScore);
+    }
+
+    public int getScoreByTeam(final Team team) {
+
+        int teamScore = 0;
+        int scoreByCypher;
+        final List<Cypher> cyphers = cypherService.getAll();
+        for (Cypher cypher: cyphers) {
+            scoreByCypher = getScoreByTeamAndCypher(team, cypher);
+            teamScore += scoreByCypher;
+        }
+        return teamScore;
+    }
```

Comment:
Urcite jsem pro radkovo reseni s operatorem `+=`.
Optional: Nejelegantnejsi metoda je pak pouziti streamu:
```java
public int getScoreByTeam(final Team team) {
    final List<Cypher> cyphers = cypherService.getAll();
    return cyphers.stream()
            .map(cypher -> getScoreByTeamAndCypher(team, cypher))
            .sum();
}

Reply:
Kdyz dam ten stream, tak mi sviti cervene .sum(), idea pise, ze to nezna. Zatim teda necham +=

Reply:
hotovo